sync/class.jetpack-sync-wp-replicastore.php

```diff
@@ -745,18 +750,32 @@ private function table_checksum( $table, $columns, $id_column, $where_sql = '1=1
 			$columns_sql = implode( ',', $sanitized_columns );
 		}
 
-		if ( $min_id !== null ) {
-			$min_id     = intval( $min_id );
-			$where_sql .= " AND $id_column >= $min_id";
-		}
+		if ( $min_id !== null && $max_id !== null ) {
+			if ( $min_id == $max_id ) {
+				$min_id = intval( $min_id );
+				$where_sql = " $id_column = $min_id LIMIT 1";
+			} else {
+				if ( $min_id !== null ) {
+					$min_id = intval( $min_id );
+					$max_id = intval( $max_id );
+					$size = $max_id - $min_id;
+					$where_sql = " $id_column >= $min_id AND $id_column <= $max_id LIMIT $size";
+				}
+			}
+		} else {
+			if ( $min_id !== null ) {
+				$min_id = intval( $min_id );
+				$where_sql .= " AND $id_column >= $min_id";
+			}
 
-		if ( $max_id !== null ) {
-			$max_id     = intval( $max_id );
-			$where_sql .= " AND $id_column <= $max_id";
+			if ( $max_id !== null ) {
+				$max_id = intval( $max_id );
+				$where_sql .= " AND $id_column <= $max_id";
+			}
 		}
 
-		$query  = <<<ENDSQL
-			SELECT CONV(BIT_XOR(CRC32(CONCAT({$columns_sql}))), 10, 16)
+		$query = <<<ENDSQL
+			SELECT CONV(BIT_XOR(CRC32(CONCAT({$columns_sql}))) + COUNT({$id_column}), 10, 16)
```

Comment:
Should we also include a test that includes a collision that the `COUNT` helps resolve?

Reply:
@enejb I'm okay adding this in a follow-up PR if you're inclined. What do you think?

Reply:
Resolved via https://github.com/Automattic/jetpack/pull/12151/files#diff-4eb5c566d5a7f44787a78bec8c57949bR337